Design/spectrum_number_and_workspace_index_abstraction.md

```diff
@@ -0,0 +1,269 @@
+# Indexing Abstraction Layer
```

Comment:
@FedeMPouzols Well, but even if the rows are not spectra we still have `MatrixWorkspace::getSpectrum()` that returns an `ISpectrum` (with data that is not a spectrum). If we really want to solve this we need new workspace types.